### DISCUSSION TITLE
Intersection of Spherical polygons

### DISCUSSION THREAD

**Discussion by hmb1** [boostorg/geometry#1021] [Q&A]
HI there,
 I am involved in Climate Research  and  have a requirement to  intersect spherical convex polygons on the unit sphere and am intending to use the boost libs.  Is there a document detailing the intersection method  that  boost geometry implements ?  ( convex or non convex )  
Thanks very much.

**Comment by awulkiew** [comment]
No, there is no document describing this, only code and comments.

I don't know what part of the algorithm is the most interesting to you so I'll describe the lowest level i.e. how the intersection of spherical segments is calculated.

By default the intersection of spherical segments is calculated with this strategy:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/strategies/spherical/intersection.hpp#L95
They are represented by 3D normal vectors of planes containing the great circles in ECEF 3D cartesian system. So both circles are intersected in 3D and then converted back to spherical.

To use it I propose to pass relate umbrella strategy, like that:
```
bg::strategies::relate::spherical<> strategy;
bg::intersection(poly1, poly2, multipoly, strategy);
```

It is also possible to use geographic strategy passing spheroid with A=B (flattening = 0) in which case the spheroidal computation is avoided. The method is the one described by Sjoberg:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/formulas/sjoberg_intersection.hpp#L33

To use it pass relate geographic strategy instead:
```
bg::strategies::relate::geographic<> strategy(bg::srs::spheroid<double>(1.0, 1.0));
bg::intersection(poly1, poly2, multipoly, strategy);
```

**Comment by hmb1** [comment]
HI Adam,
  Many thanks for the links - looks like the intersection is pretty complex business 
 ...Henry
